### PR TITLE
Add cursor support to axiom-rs

### PR DESCRIPTION
**Pull request by damaredayo**
Fixes DX-449

### CODE REVIEW

File: src/datasets/model.rs [hunk 1]
```diff
@@ -320,6 +320,8 @@ pub struct Query {
     pub apl: String,
     pub start_time: Option<DateTime<Utc>>,
     pub end_time: Option<DateTime<Utc>>,
+    pub cursor: Option<String>,
+    pub include_cursor: Option<bool>,
```
Comment (bahlo) [inline src/datasets/model.rs:324]: Can we make this a non-option?

File: src/datasets/model.rs [hunk 2]
```diff
@@ -339,6 +341,11 @@ pub struct QueryOptions {
     pub start_time: Option<DateTime<Utc>>,
     // The end time of the query.
     pub end_time: Option<DateTime<Utc>>,
+    // The cursor for use in pagination.
+    pub cursor: Option<String>,
+    // include_cursor specifies whether the event that matches the cursor should be
```
Comment (bahlo) [inline src/datasets/model.rs:346]:
```suggestion
    // Specifies whether the event that matches the cursor should be
```

File: tests/cursor.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
+use async_trait::async_trait;
```
Comment (bahlo) [inline tests/cursor.rs:1]: Can you move this into the files at the bottom with the other tests?

**Review by bahlo — APPROVED** [review]
LGTM otherwise, rustfmt is mad at you though

File: src/datasets/model.rs (hunk 1 above)
Comment (damaredayo) [reply src/datasets/model.rs]: Fixed in https://github.com/axiomhq/axiom-rs/pull/44/commits/ae957d05e8b4de06920875ca85746ab7b6336391

**Pull request merged**